src/main/java/net/javadiscord/javabot/systems/starboard/StarboardManager.java

```diff
@@ -137,13 +141,18 @@ private void addMessageToStarboard(Message message, int stars, @NotNull Starboar
 				.sendMessage(String.format("%s %s", config.getEmojis().get(0), stars))
 				.setActionRow(Button.link(message.getJumpUrl(), "Jump to Message"))
 				.setEmbeds(embed);
-		for (Message.Attachment a : message.getAttachments()) {
-			try {
-				action.addFiles(FileUpload.fromData(a.getProxy().download().get(), a.getFileName()));
-			} catch (InterruptedException | ExecutionException e) {
-				action.addContent("Could not add Attachment: " + a.getFileName());
+		List<Attachment> attachments = message.getAttachments();
+		
+		if(attachments.size() > 1) {
+			for (Message.Attachment a : attachments) {
+				try {
+					action.addFiles(FileUpload.fromData(a.getProxy().download().get(), a.getFileName()));
```

Comment:
If possible, please don't download and re-upload the image but instead use the existing URL.

Also, don't block the thread for network requests.

Reply:
Actually the file upload is **not** my code. I just added the `if` here

Reply:
> If possible, please don't download and re-upload the image but instead use the existing URL.
> 
> Also, don't block the thread for network requests.

Also I just checked. It's not possible to NOT reupload an attachment if you want to include it along a embed. If you just send an URL Discord won't convert it to an image.
![image](https://github.com/Java-Discord/JavaBot/assets/66302346/1abdcb82-24b2-4d36-8d28-a571e316a76c)